gfx.h

```diff
@@ -191,15 +191,23 @@ GfxRaytracingPrimitive const *gfxAccelerationStructureGetRaytracingPrimitives(Gf
 
 class GfxRaytracingPrimitive { GFX_INTERNAL_NAMED_HANDLE(GfxRaytracingPrimitive); public: };
 
+enum GfxRaytracingPrimitiveFlags
+{
+    GfxRaytracingPrimitiveFlags_None = 0,
+    GfxRaytracingPrimitiveFlags_NonOpaque = 1,
+
+    GfxRaytracingPrimitiveFlags_Count
+};
+
 GfxRaytracingPrimitive gfxCreateRaytracingPrimitive(GfxContext context, GfxAccelerationStructure acceleration_structure);
 GfxResult gfxDestroyRaytracingPrimitive(GfxContext context, GfxRaytracingPrimitive raytracing_primitive);
 
-GfxResult gfxRaytracingPrimitiveBuild(GfxContext context, GfxRaytracingPrimitive raytracing_primitive, GfxBuffer vertex_buffer, uint32_t vertex_stride = 0);
-GfxResult gfxRaytracingPrimitiveBuild(GfxContext context, GfxRaytracingPrimitive raytracing_primitive, GfxBuffer index_buffer, GfxBuffer vertex_buffer, uint32_t vertex_stride = 0);
+GfxResult gfxRaytracingPrimitiveBuild(GfxContext context, GfxRaytracingPrimitive raytracing_primitive, GfxBuffer vertex_buffer, uint32_t vertex_stride = 0, GfxRaytracingPrimitiveFlags flags = GfxRaytracingPrimitiveFlags_NonOpaque);
```

Comment:
The type should be `uint32_t` since you can combine multiple flags possibly (although a single one exists atm 😛).

So either `uint32_t build_flags = 0` or `uint32_t build_flags = kGfxBuildRaytracingPrimitiveFlag_Opaque` depending on what we'd like the default behaviour to be.
More leaning towards `0` being the default atm.

Reply:
Ive set this as default `0`. There are performance advantages to setting geometry as opaque which is why it could be useful as the default however then you have the potential issue of alpha masked geometry not being detected if not explicitly set to `0`. So either way the user should use the correct flag for best results (and a default '0' is less code)